### PR TITLE
fix(RuleEntity) : add missing source 'GLPI Native Inventory'

### PR DESCRIPTION
**Pull request by stonebuzz**
Add missing source ```GLPI Native Inventory``` from ```RuleEntity```

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !26418

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

File: src/RuleImportEntity.php [hunk 1]
```diff
@@ -192,13 +192,15 @@ public function getAdditionalCriteriaDisplayPattern($ID, $condition, $pattern)
 
         $crit = $this->getCriteria($ID);
         if (count($crit) && $crit['field'] == '_source') {
-            if ($pattern != 'GLPI') {
+            if ($pattern != 'GLPI' && $pattern != 'NATIVE_INVENTORY') {
+                $name = $pattern;
+            } if ($pattern == 'NATIVE_INVENTORY') {
```
Comment (cedric-anne) [inline src/RuleImportEntity.php:197]:
```suggestion
            } elseif ($pattern == 'NATIVE_INVENTORY') {
```

Comment (trasher) [reply src/RuleImportEntity.php:197]: :+1:

File: src/RuleImportEntity.php [hunk 2]
```diff
@@ -151,7 +151,7 @@ public function displayAdditionalRuleCondition($condition, $criteria, $name, $va
         global $PLUGIN_HOOKS, $CFG_GLPI;
 
         if ($criteria['field'] == '_source') {
-            $tab = ['GLPI' => __('GLPI')];
+            $tab = ['GLPI' => __('GLPI'), 'NATIVE_INVENTORY' => AutoUpdateSystem::NATIVE_INVENTORY ];
```
Comment (cedric-anne) [inline src/RuleImportEntity.php:154]: Maybe this sting should be translated.

Comment (stonebuzz) [reply src/RuleImportEntity.php:154]: done

Comment (cedric-anne) [inline src/RuleImportEntity.php:154]:
```suggestion
            $tab = ['GLPI' => __('GLPI'), 'NATIVE_INVENTORY' => AutoUpdateSystem::getLabelFor(AutoUpdateSystem::NATIVE_INVENTORY)];
```

File: src/AutoUpdateSystem.php [hunk 3]
```diff
@@ -41,4 +41,14 @@ public static function getTypeName($nb = 0)
     {
         return _n('Update Source', 'Update Sources', $nb);
     }
+
+    public static function getLabelFor($key)
```
Comment (cedric-anne) [inline src/AutoUpdateSystem.php:45]:
```suggestion
    public static function getLabelFor(string $key): string
```

**Review by cedric-anne — APPROVED** [review]

File: src/RuleImportEntity.php [hunk 4]
```diff
@@ -192,13 +192,15 @@ public function getAdditionalCriteriaDisplayPattern($ID, $condition, $pattern)
 
         $crit = $this->getCriteria($ID);
         if (count($crit) && $crit['field'] == '_source') {
-            if ($pattern != 'GLPI') {
+            if ($pattern != 'GLPI' && $pattern != 'NATIVE_INVENTORY') {
+                $name = $pattern;
```
Comment (cedric-anne) [inline src/RuleImportEntity.php:196]: I misread the diff. To not break previous behaviour, code should be:
```suggestion
            if ($pattern == 'GLPI') {
                $name = $pattern;
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**